src/ntangle.nim

```diff
@@ -400,20 +400,24 @@ proc getHeaderArgs(file: string, line: string, lnum: int): LangAndArgs =
     if startHeaderArgs >= 2:
       headerArgsRaw = spaceSepParts[startHeaderArgs .. spaceSepParts.high]
   if haType != haNone:
-    #echo headerArgsRaw
+    var lastKey = false
+    var keyVal: tuple[key, val: string]
+    template notEmpty(kv: untyped): untyped = kv.key.len > 0 or kv.val.len > 0
     for i, h in headerArgsRaw:
-      if h.len >= 2 and h[0] == ':':
-        if i == headerArgsRaw.high:
-          raise newException(OrgError, fmt("The header args are ending with a ':key' which is not valid. Found {headerArgsRaw}"))
-        if headerArgPair != "":
-          headerArgs.add(headerArgPair)
-        headerArgPair = h[1 .. h.high]
-        #echo fmt"{i} - {headerArgPair}"
+      if h.len >= 2 and h[0] == ':': # this is a key
+        if notEmpty(keyVal): # add current `keyVal` if anything
+          headerArgs.add keyVal
+          keyVal = (key: "", val: "")
+        keyVal.key = h[1 .. h.high]
+        lastKey = true
       else:
-        headerArgPair = headerArgPair & " " & h.strip(chars = {'"'})
```

Comment:
@Vindaar ok.. this part got left out: 

```
h.strip(chars = {'"'})
```

Reply:
Ohh, thanks. Guess I wasn't paying attention, oops!